.github/workflows/ci.yml

```diff
@@ -24,21 +24,18 @@ jobs:
             ruby: 2.7
           - gemfile: rails-main.gemfile
             ruby: 3.0
-
+    env:
+      BUNDLE_GEMFILE: gemfiles/${{ matrix.gemfile }}
```

Comment:
We'll see if this change cleans it up or not 

https://github.com/ruby/setup-ruby?tab=readme-ov-file#matrix-of-gemfiles